src/effects/effectsmanager.h

```diff
@@ -25,6 +25,26 @@ class EffectManifest;
 
 typedef QMap<EffectManifestParameter::ParameterType, QList<EffectParameterPointer>> ParameterMap;
 
+// EffectsManager is the interface between the parts of the effects system in
+// the main thread and the rest of Mixxx. It creates/destroys a fixed
+// set of EffectChainSlots on Mixxx startup/shutdown. EffectManager uses
+// EffectBackends and EffectManifests to create EffectProcessors. The
+// EffectProcessors are sent down to the EffectChainSlots, then down to the
+// EffectSlots which use the EffectProcessors to create EngineEffects and add
+// them to the audio engine.
```

Comment:
this describes well what happens, but doesn't give any background on why - what are the "real-world" implications of this code? Maybe I am also expecting too much though, just a thought.

Reply:
Interesting comment. I'll have to think about that...

Reply:
any new thoughts? ;)

Reply:
I had trouble explaining the "why" behind this because the reason was bad! That reason was that the EffectManager class had way too many responsibilities. Splitting it up into many small classes enabled a straightforward path through the code for loading effects.